src/w3d/renderer/render2dsentence.cpp

```diff
@@ -42,13 +52,39 @@ FontCharsClass::FontCharsClass() :
     m_gdiFont(nullptr),
     m_gdiBitmapBits(nullptr),
     m_memDC(nullptr),
+#endif
+#ifdef BUILD_WITH_FREETYPE
+    m_ftLibrary(nullptr),
+    m_ftFace(nullptr),
 #endif
     m_unicodeCharArray(nullptr),
     m_firstUnicodeChar(0xFFFF),
     m_lastUnicodeChar(0),
     m_isBold(false)
 {
     std::memset(m_asciiCharArray, 0, sizeof(m_asciiCharArray));
+    // Use Fontconfig to locate system fonts
+#ifdef BUILD_WITH_FONTCONFIG
+    if (FcInit() == FcTrue) {
+        m_fc = FcInitLoadConfigAndFonts();
+        if (m_fc == nullptr) {
+            captainslog_error("Failed to load Fontconfig config & fonts ");
+        }
+    } else {
+        captainslog_error("Failed to init Fontconfig");
+    }
+#endif
+    // Use Freetype to loader & rasterize fonts
+#ifdef BUILD_WITH_FREETYPE
+    FT_Error error = FT_Init_FreeType(&m_ftLibrary);
+    if (error != FT_Err_Ok) {
+#ifdef FT_CONFIG_OPTION_ERROR_STRINGS
```

Comment:
Well how about:

```c++
void LogFtError(const char *text, FT_Error err)
{
#ifdef FT_CONFIG_OPTION_ERROR_STRINGS
  captainslog_error("%s: %s", text, FT_Error_String(error));
#else
  captainslog_error(text);
#endif
}
```
This would get rid of bulky code.

Reply:
Would be nice. All this `#ifdef` here is difficult to look at.